Add menu-driven approch

Ank1taS opened the pull request:
Approach to menu-driven :
Initially, the `filter options` will be displayed along with the `quit option` and prompt the user for his/her option.
If the user gives a valid filter option or quit option then it will work accordingly, but if multiple options or an invalid option is provided then the menu will display and prompt for the user option in a loop until getting a valid one 

**_What has been done by far_**
only the **menu display and option validation** part is done.

_**To do**_
Further **command-line arguments validations, input file validation, output file validation and appropriate filter function calls based on user options** will be done.

Ank1taS commented:
What has been done by far
- command-line arguments validations, 
- menu display and option validation.
- input file validation, 
- output file validation
- appropriate filter function calls based on user options.

All required changes in `main.cpp` are done to make the application **menu driven/user-friendly** so far

Himanshu40 commented on main.cpp:
```diff
@@ -7,39 +8,63 @@
+            
+            // check validates option
+            if (filtersOptions.find(filter) != std::string::npos) {
+                std::cout << "Good choice" << std::endl;
```
Can we have more descriptive information instead of `Good Choice` ? @chandrakant100 @Ank1taS Otherwise everything is fine from my side.

Ank1taS replied:
How about `Applied filter on [File name] successfully`? But in that case, the code will be more extended if we want to specify which filter is applied. Alternatively, we can specify the same statement in the switch block in each case as ` Applied [filter name] filter on [file name] successfully`.

Himanshu40 replied:
`Applied [filter name] filter on [file name] successfully` will be the correct one to describe what happened to the file.

Ank1taS replied:
ok I will update pr

chandrakant100 replied:
> `Applied [filter name] filter on [file name] successfully` will be the correct one to describe what happened to the file.

This sounds more friendly

Ank1taS commented:
What changes were done in the above commit :
- replaced `good choice` massage with appropriate massage with a specific filter name.
- Updated usage massage (Removed flag as we don't prompt any flag along with execute statement).
- Rearrange file validation as it should be done before prompting the filter option.
